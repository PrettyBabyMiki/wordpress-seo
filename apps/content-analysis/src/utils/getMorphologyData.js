/* eslint-disable global-require */
// Disabling global require to be able to fail.

import { merge } from "lodash-es";
let morphologyData = null;

/**
 * Loads morphology data from disk, if available.
 * @returns {Object} The morphology data, or an empty object if not available.
 */
function loadLocalMorphologyData() {
	let data, dataDe, dataNL, dataES, dataFR, dataRU, dataIT, dataPT, dataID, dataPL, dataAR, dataSV, dataHE, dataHU,
		dataNB, dataTR, dataCS, dataSK, dataEL, dataJA = {};
	try {
		data = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-en-v4.json" );
		dataDe = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-de-v9.json" );
		dataNL = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-nl-v9.json" );
		dataES = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-es-v9.json" );
		dataFR = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-fr-v9.json" );
		dataRU = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-ru-v10.json" );
		dataIT = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-it-v10.json" );
		dataPT = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-pt-v9.json" );
		dataID = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-id-v9.json" );
		dataPL = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-pl-v9.json" );
		dataAR = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-ar-v9.json" );
		dataSV = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-sv-v1.json" );
		dataHE = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-he-v1.json" );
		dataHU = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-hu-v2.json" );
		dataNB = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-nb-v1.json" );
		dataTR = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-tr-v1.json" );
		dataCS = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-cs-v1.json" );
		dataSK = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-sk-v1.json" );
		dataEL = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-el-v1.json" );
		dataJA = require( "../../../../packages/yoastseo/premium-configuration/data/morphologyData-ja-v1.json" );
	} catch ( error ) {
		// Falling back to empty data.
	}
	return merge( data, dataDe, dataNL, dataES, dataFR, dataRU, dataIT, dataPT, dataID, dataPL, dataAR, dataSV, dataHE, dataHU, dataNB, dataTR,
		dataCS, dataSK, dataEL, dataJA );
}

/**
 * Get morphology data. To be used in the analysis to recognize different word forms.
 * @returns {Object} The morphology data.
 */
export default function getMorphologyData() {
	if ( morphologyData === null ) {
		morphologyData = loadLocalMorphologyData();
	}
	return morphologyData;
}
