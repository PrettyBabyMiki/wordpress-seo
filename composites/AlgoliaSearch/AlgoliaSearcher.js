/* External dependencies */
import React from "react";
import initAlgoliaSearch from "algoliasearch";
import isUndefined from "lodash/isUndefined";
import { speak as a11ySpeak } from "@wordpress/a11y";
import PropTypes from "prop-types";
import styled from "styled-components";
import { __ } from "@wordpress/i18n";

/* Internal dependencies */
import Loading from "./Loading";
import SearchBar from "./SearchBar";
import SearchResultDetail from "./SearchResultDetail";
import SearchResults from "./SearchResults";

const AlgoliaSearchWrapper = styled.div`
	margin: 0 auto 20px auto;
	box-sizing: border-box;
`;

const VIEW = {
	SEARCH: "SEARCH",
	DETAIL: "DETAIL",
};

/**
 * The AlgoliaSearcher component.
 */
class AlgoliaSearcher extends React.Component {
	/**
	 * AlgoliaSearcher constructor.
	 *
	 * @constructor
	 * @param {object} props Properties of the AlgoliaSearcher component.
	 *
	 * @returns {void}
	 */
	constructor( props ) {
		super( props );

		this.state = {
			searchString: "",
			searching: false,
			results: [],
			errorMessage: "",
			usedQueries: {},
			currentDetailViewIndex: -1,
			currentView: VIEW.SEARCH,
		};

		this.initAlgoliaClient();
	}

	/**
	 * Initializes the algolia client and index variables.
	 *
	 * @returns {void}
	 */
	initAlgoliaClient() {
		this.client = initAlgoliaSearch( this.props.algoliaApplicationId, this.props.algoliaApiKey );
		this.index = this.client.initIndex( this.props.algoliaIndexName );
	}

	/**
	 * Handles the form submit event. Stores the search string and performs a search.
	 *
	 * @param {string} searchString The entered search string.
	 *
	 * @returns {void}
	 */
	onSearch( searchString ) {
		if ( searchString === "" ) {
			return;
		}
		const usedQueries = this.addUsedQuery( searchString );

		if ( this.props.onQueryChange ) {
			this.props.onQueryChange( usedQueries );
		}

		// Updating the state will re-render the whole component.
		this.setState( {
			searchString,
			usedQueries: usedQueries,
			searching: true,
		}, this.updateSearchResults );
	}

	/**
	 * Adds a search string to the list of used queries.
	 *
	 * @param {string} searchString The search string that was used in the last search.
	 *
	 * @returns {Object} The usedQueries state object.
	 */
	addUsedQuery( searchString ) {
		const usedQueries = this.state.usedQueries;

		if ( isUndefined( usedQueries[ searchString ] ) ) {
			usedQueries[ searchString ] = {};
		}

		return usedQueries;
	}

	/**
	 * Processes the passed search error.
	 *
	 * @param {object} error The search error to process.
	 *
	 * @returns {void}
	 */
	processSearchError( error ) {
		this.setState( {
			errorMessage: error.message,
			searching: false,
		} );
	}

	/**
	 * Processes the search results.
	 *
	 * @param {object} results The results to process.
	 *
	 * @returns {void}
	 */
	processResults( results ) {
		this.setState( {
			results: results,
			errorMessage: "",
			searching: false,
		} );
	}

	/**
	 * Performs a search with the search string saved in the state.
	 *
	 * @returns {void}
	 */
	updateSearchResults() {
		this.performSearch( this.state.searchString )
			.then( this.processResults.bind( this ) )
			.catch( this.processSearchError.bind( this ) );
	}

	/**
	 * Performs a search with the given search string within the Algolia index.
	 *
	 * @param {string} searchString The words or sentence to get the results for.
	 *
	 * @returns {Promise} The promise that is performing the search.
	 */
	performSearch( searchString ) {
		return new Promise( ( resolve, reject ) => {
			this.index.search( searchString, ( err, data ) => {
				if ( err ) {
					reject( err );
					return;
				}

				resolve( data.hits );
			} );
		} );
	}

	/**
	 * Adds the data from the clicked result to the information associated with the current search string.
	 *
	 * @param {number} currentDetailViewIndex The current detail view index.
	 *
	 * @returns {Object} Object containing the currently used queries with additional post data.
	 */
	addResultToUsedQueries( currentDetailViewIndex ) {
		const post = this.state.results[ currentDetailViewIndex ];
		const usedQueries = this.state.usedQueries;

		usedQueries[ this.state.searchString ][ post.objectID ] = {
			title: post.post_title, // eslint-disable-line
			link: post.permalink,
		};

		return usedQueries;
	}

	/**
	 * Sets the current view to the detail view and set the article index.
	 *
	 * @param {number} currentDetailViewIndex The index of the article you want to show in the state.results array.
	 *
	 * @returns {void}
	 */
	showDetailView( currentDetailViewIndex ) {
		const usedQueries = this.addResultToUsedQueries( currentDetailViewIndex );

		this.setState( {
			currentDetailViewIndex,
			usedQueries,
			currentView: VIEW.DETAIL,
		} );
	}

	/**
	 * Sets the current view to the search view.
	 *
	 * @returns {void}
	 */
	hideDetailView() {
		this.setState( { currentView: VIEW.SEARCH } );
	}

	/**
	 * Creates the Search Bar component, unless you are in the detail view.
	 *
	 * @returns {ReactElement} Searchbar component.
	 */
	createSearchBar() {
		return (
			<SearchBar
				submitAction={ this.onSearch.bind( this ) }
				searchString={ this.state.searchString }
				enableLiveSearch={ this.props.enableLiveSearch }
			/>
		);
	}

	/**
	 * Gets the error message.
	 *
	 * @returns {ReactElement} Error component.
	 */
	getErrorMessage() {
		const errorMessage = __( "Something went wrong. Please try again later.", "yoast-components" );

		a11ySpeak( errorMessage );

		return (
			<p>{ errorMessage }</p>
		);
	}

	/**
	 * Gets the loading indicator.
	 *
	 * @returns {ReactElement} Loader component.
	 */
	getLoadingIndicator() {
		const loadingPlaceholder = __( "Loading...", "yoast-components" );

		return <Loading placeholder={ loadingPlaceholder } />;
	}

	/**
	 * Gets the search results.
	 *
	 * @returns {ReactElement} Search results component.
	 */
	getSearchResults() {
		return <SearchResults
			{ ...this.props }
			searchString={ this.state.searchString }
			results={ this.state.results }
			onClick={ this.showDetailView.bind( this ) }
		/>;
	}

	/**
	 * Determines what the search results view needs to look like.
	 *
	 * @returns {ReactElement} Returns a specific search result component based on state.
	 */
	determineSearchResultsView() {
		// Show error when a error message is set.
		if ( this.state.errorMessage ) {
			return this.getErrorMessage();
		}

		// Show loading indicator when search results are being fetched and no results are present.
		if ( this.state.searching && ! this.state.results ) {
			return this.getLoadingIndicator();
		}

		// Show the list of search results if the postId for the detail view isn't set.
		return this.getSearchResults();
	}

	/**
	 * Gets a specific post from the list of results.
	 *
	 * @param {number} index The index of the post to retrieve.
	 *
	 * @returns {Object} The post object associated with the index.
	 */
	getPostFromResults( index ) {
		return this.state.results[ index ];
	}

	/**
	 * Gets the search view.
	 *
	 * @returns {ReactElement} Search view component.
	 */
	getSearchView() {
		return (
			<AlgoliaSearchWrapper
				innerRef={ ( el ) => {
					this.searchViewWrapper = el;
				} }
			>
				{ this.createSearchBar() }
				{ this.determineSearchResultsView() }
			</AlgoliaSearchWrapper>
		);
	}

	/**
	 * Gets the search result detail.
	 *
	 * @returns {ReactElement} Detail view component.
	 */
	getDetailView() {
		return (
			<AlgoliaSearchWrapper>
				<SearchResultDetail
					{ ...this.props }
					post={ this.getPostFromResults( this.state.currentDetailViewIndex ) }
					onBackButtonClicked={ this.hideDetailView.bind( this ) }
				/>
			</AlgoliaSearchWrapper>
		);
	}

	/**
	 * Move focus back to the clicked link when going back from Detail to Search view.
	 *
	 * Call this function on componentDidUpdate() to avoid it runs on first rendering.
	 *
	 * @returns {void}
	 */
	moveFocusBackToClickedSearchResult() {
		const clickedLinkIndex = this.state.currentDetailViewIndex;
		// When is search view and a search results has been previously clicked.
		if ( this.state.currentView === "SEARCH" && clickedLinkIndex >= 0 ) {
			const resultLinks = this.searchViewWrapper.querySelectorAll( "ul a" );

			if ( ! resultLinks.length ) {
				return;
			}

			resultLinks[ clickedLinkIndex ].focus();
		}
	}

	/**
	 * Renders the React component.
	 *
	 * Called upon each state/props change. Determines and renders the view to render.
	 *
	 * @returns {ReactElement} The content of the component.
	 */
	render() {
		switch ( this.state.currentView ) {
			case VIEW.SEARCH:
				return this.getSearchView();
			case VIEW.DETAIL:
				return this.getDetailView();
		}
	}

	/**
	 * Moves focus back to the clicked search results if this component updates.
	 *
	 * Prevents focus loss.
	 *
	 * @returns {void}
	 */
	componentDidUpdate() {
		this.moveFocusBackToClickedSearchResult();
	}
}

AlgoliaSearcher.propTypes = {
	algoliaApplicationId: PropTypes.string,
	algoliaApiKey: PropTypes.string,
	algoliaIndexName: PropTypes.string,
	onQueryChange: PropTypes.func,
	enableLiveSearch: PropTypes.bool,
};

AlgoliaSearcher.defaultProps = {
	algoliaApplicationId: "RC8G2UCWJK",
	algoliaApiKey: "459903434a7963f83e7d4cd9bfe89c0d",
	algoliaIndexName: "knowledge_base_all",
	enableLiveSearch: false,
};

export default AlgoliaSearcher;
