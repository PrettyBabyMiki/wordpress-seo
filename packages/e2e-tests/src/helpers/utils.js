/**
 * WordPress e2e utilities
 */
import {
    installPlugin,
    activatePlugin,
    visitAdminPage,
    pressKeyWithModifier,
} from "@wordpress/e2e-test-utils";

import { addQueryArgs } from '@wordpress/url';

export const createCustomPostType = async( slug ) => {
   
}