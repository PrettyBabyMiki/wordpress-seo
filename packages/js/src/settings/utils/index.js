import { filter, isArray, isObject, join, reduce } from "lodash";

/**
 * Determines if the given value should flatten.
 * @param {*} value The value.
 * @returns {boolean} True if the value is an object or an array.
 */
const shouldFlatten = value => isObject( value ) || isArray( value );

/**
 * Flattens nested object to a single level. Props will be keyed by dot syntax path.
 * @param {Object} object The object to flatten.
 * @param {string} parentPath The parent object path.
 * @param {function} getShouldFlatten Determines whether the entry should be flattened.
 * @returns {Object} Flattened object.
 */
export const flattenObject = ( object, parentPath = "", getShouldFlatten = shouldFlatten ) => reduce(
	object,
	( acc, value, key ) => {
		const flatKey = join( filter( [ parentPath, key ], Boolean ), "." );

		return getShouldFlatten( value, key ) ? {
			...acc,
			...flattenObject( value, flatKey, getShouldFlatten ),
		} : {
			...acc,
			[ flatKey ]: value,
		};
	},
	{}
);
