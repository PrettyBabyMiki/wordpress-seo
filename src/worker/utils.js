/**
 * Encodes the payload for transfer.
 *
 * @param {Object} payload The payload.
 *
 * @returns {string} Transferable string.
 */
export function encodePayload( payload ) {
	return JSON.stringify( payload );
}

/**
 * Decodes the payload for use.
 *
 * @param {string} payload The encoded payload.
 *
 * @returns {any} The parsed result.
 */
export function decodePayload( payload ) {
	try {
		return JSON.parse( payload );
	} catch( error ) {
		return payload;
	}
}
