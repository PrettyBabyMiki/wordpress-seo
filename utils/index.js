import { makeOutboundLink } from "./makeOutboundLink";
import { getFeed } from "./getFeed";

export default {
	getFeed,
	makeOutboundLink,
};
