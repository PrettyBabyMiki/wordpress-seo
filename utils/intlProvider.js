import React from "react";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";

export const createComponentWithIntl = ( children, props = { locale: "en" }, renderOptions = {} ) => {
	return renderer.create(
		<IntlProvider { ...props }>
			{ children }
		</IntlProvider>,
		renderOptions
	);
};
