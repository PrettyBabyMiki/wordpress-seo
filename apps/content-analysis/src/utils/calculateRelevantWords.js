import { get, isEqual, take } from "lodash-es";
import getLanguage from "yoastsrc/helpers/getLanguage";
import getWords from "yoastsrc/stringProcessing/getWords";
import {
	collapseRelevantWordsOnStem,
	getRelevantWords,
	getRelevantWordsFromPaperAttributes,
	getRelevantCombinations,
	sortCombinations,
} from "yoastsrc/stringProcessing/relevantWords";
import { getSubheadingsTopLevel, removeSubheadingsTopLevel } from "yoastsrc/stringProcessing/getSubheadings";
import { retrieveAbbreviations } from "../../../../packages/yoastseo/src/stringProcessing/relevantWords";
import getMorphologyData from "./getMorphologyData";

const morphologyData = getMorphologyData();

// Cache the relevant words.
let previousRelevantWordsInternalLinking = {
	text: "",
	locale: "en_US",
	description: "",
	keyword: "",
	synonyms: "",
	title: "",
	data: {},
};

let previousRelevantWordsInsights = {
	text: "",
	locale: "en_US",
	data: {},
};

/**
 * Rounds number to four decimals.
 *
 * @param {number} number The number to be rounded.
 *
 * @returns {number} The rounded number.
 */
function formatNumber( number ) {
	if ( Math.round( number ) === number ) {
		return number;
	}

	return Math.round( number * 10000 ) / 10000;
}

/**
 * Calculates all properties for the relevant word objects.
 *
 * @param {Paper}   paper           The paper to analyse.
 * @param {boolean} useAttributes   Whether text from attributes should also be considered for relevant words.
 *
 * @returns {Object} The relevant word objects.
 */
function calculateRelevantWords( paper, useAttributes ) {
	let text = paper.text;
	const language = getLanguage( paper.locale );
	const languageMorphologyData = get( morphologyData, language, false );

	const subheadings = getSubheadingsTopLevel( text ).map( subheading => subheading[ 2 ] );

	const attributes = useAttributes ? [
		paper.keyword,
		paper.synonyms,
		paper.description,
		paper.title,
		subheadings.join( " " ),
	] : [];

	text = useAttributes ? removeSubheadingsTopLevel( text ) : text;

	const abbreviations = retrieveAbbreviations( text.concat( attributes.join( " " ) ) );

	const relevantWordsFromText = getRelevantWords( text, abbreviations, language, languageMorphologyData );

	let relevantWordsFromPaperAttributes = [];

	if ( useAttributes ) {
		relevantWordsFromPaperAttributes = getRelevantWordsFromPaperAttributes(
			attributes,
			abbreviations,
			language,
			languageMorphologyData
		);

		/*
		 * Analogous to the research src/researches/relevantWords.js, all relevant words that come from paper attributes
		 * (and not from text) get a times-3 number of occurrences to support the idea that they are more important than
		 * the words coming from the text. For instance, if a word occurs twice in paper attributes it receives
		 * number_of_occurrences = 6.
		 */
		relevantWordsFromPaperAttributes.forEach( relevantWord => relevantWord.setOccurrences( relevantWord.getOccurrences() * 3 ) );
	}

	const collapsedWords = collapseRelevantWordsOnStem( relevantWordsFromPaperAttributes.concat( relevantWordsFromText ) );
	sortCombinations( collapsedWords );

	/*
	 * Analogous to the research src/researches/relevantWords.js, we limit the number of relevant words in consideration
	 * to 100, i.e. we take 100 first relevant words from the list sorded by number of occurrences first and then
	 * alphabetically.
	 */
	const relevantWords = take( getRelevantCombinations( collapsedWords ), 100 );

	return relevantWords.map( ( word ) => {
		return {
			word: word.getWord(),
			stem: word.getStem(),
			occurrences: word.getOccurrences(),
		};
	} );
}

/**
 * Retrieves the relevant words for Internal linking. Uses cached version when possible.
 *
 * @param {Paper} paper   The paper to get relevant words for.
 *
 * @returns {Object} The relevant words.
 */
export function relevantWordsForInternalLinking( paper ) {
	const text = paper.text;
	const locale = paper.locale;
	const description = paper.description;
	const keyword = paper.keyword;
	const synonyms = paper.synonyms;
	const title = paper.title;

	if (
		! isEqual( text, previousRelevantWordsInternalLinking.text ) ||
		! isEqual( locale, previousRelevantWordsInternalLinking.locale ) ||
		! isEqual( description, previousRelevantWordsInternalLinking.description ) ||
		! isEqual( keyword, previousRelevantWordsInternalLinking.keyword ) ||
		! isEqual( synonyms, previousRelevantWordsInternalLinking.synonyms ) ||
		! isEqual( title, previousRelevantWordsInternalLinking.title )
	) {
		previousRelevantWordsInternalLinking = {
			text,
			locale,
			description,
			keyword,
			synonyms,
			title,
			data: calculateRelevantWords( paper, true ),
		};
	}
	return previousRelevantWordsInternalLinking.data;
}

/**
 * Retrieves the relevant words for Insights (takes into consideration only the text itself, not attributes).
 * Uses cached version when possible.
 *
 * @param {Paper} paper   The paper to get relevant words for.
 *
 * @returns {Object} The relevant words.
 */
export function relevantWordsForInsights( paper ) {
	const text = paper.text;
	const locale = paper.locale;

	if (
		! isEqual( text, previousRelevantWordsInsights.text ) ||
		! isEqual( locale, previousRelevantWordsInsights.locale )
	) {
		previousRelevantWordsInsights = {
			text,
			locale,
			data: calculateRelevantWords( paper, false ),
		};
	}
	return previousRelevantWordsInsights.data;
}

