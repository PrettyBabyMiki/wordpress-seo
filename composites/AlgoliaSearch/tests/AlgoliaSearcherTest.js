import React from "react";
import { createComponentWithIntl } from "../../../utils/intlProvider";
import AlgoliaSearcher from "../AlgoliaSearcher.js";
jest.mock( "svg-url-loader!../../icons/search.svg", () => {
	return "mockedSvg";
} );


test( "the AlgoliaSearcher component with headingText matches the snapshot", () => {
	const component = createComponentWithIntl(
		<AlgoliaSearcher />
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
