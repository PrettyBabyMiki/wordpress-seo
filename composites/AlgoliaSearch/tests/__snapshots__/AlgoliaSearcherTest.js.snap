// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the AlgoliaSearcher component with headingText matches the snapshot 1`] = `
<div
  className="sc-kgoBCf ijaDUj"
>
  <div
    className="sc-htpNat ednHKJ"
    role="search"
  >
    <h2
      className="sc-bxivhb cptdIu"
    >
      Search the Yoast knowledge base
    </h2>
    <form
      onSubmit={[Function]}
    >
      <label
        className="sc-ifAKCX LtJIk"
        htmlFor="kb-search-input"
      >
        <svg
          aria-hidden="true"
          className="sc-kGXeez fVArXq"
          focusable="false"
          role="img"
        />
        <span
          className="screen-reader-text"
        >
          Search the Yoast knowledge base
        </span>
      </label>
      <input
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        className="sc-EHOje aywLs"
        defaultValue=""
        id="kb-search-input"
        name="search-input"
        onChange={[Function]}
        placeholder="Search the knowledge base"
        spellCheck="false"
        type="text"
      />
      <button
        className="sc-bwzfXH eeZBlJ sc-bdVaJa bsIiYo"
        onClick={undefined}
        type="button"
      >
        <span>
          Search
        </span>
      </button>
    </form>
  </div>
</div>
`;
